### PR TITLE
Html2pdf testing

### LINKED ISSUE #133: 84836 - [Newsroom] Common Modules -News Posting Article-Exported PDF links are not coming inline  for the franklin exported page.
**Severity 3 - Medium**
**Steps to Reproduce: | Analysis & Proposed Solution: | Location of the Element:**

1.open both the below URL legacy and franklin one

**Sample Reference/s:
Legacy newsroom page:**
https://newsroom.accenture.com/news/accenture-completes-acquisition-of-anser-advisory.htm
**Adobe Franklin migrated page:** 
https://main--accenture-newsroom--hlxsites.hlx.live/news/2023/accenture-completes-acquisition-of-anser-advisory

2. click on the down press release button on the right side.

3.Open the downloaded PDF file.

4.Observe on the article para the inline links are not coming in line they are little downwards

Actual: it is not coming inline

Expected: it should come inline not downwards

image
Azdo link - https://dev.azure.com/accenturecio26/WebContentDigAssetMgmt_PR1093/_workitems/edit/84836
CC: @ravkiran  @amol-anand  @raymond-h-dev

### PR DESCRIPTION
**Pull request by amol-anand**
Changed PDF generation to include links and fixed some alignment issues

Fix #133 

Test URLs:
- Before: https://main--accenture-newsroom--hlxsites.hlx.page/news/2023/rashi-test-page
- After: https://html2pdf-testing--accenture-newsroom--hlxsites.hlx.page/news/2023/rashi-test-page

@jjjaspher you were on the right track with html2pdf instead of js2pdf. Did some minor changes to make things look better. please test it and let me know if this works.

### CODE REVIEW

File: blocks/aside/aside.js [hunk 1]
```diff
@@ -165,11 +159,14 @@ export default async function decorate(block) {
...
-      if (window.jspdf) {
-        await generatePDF(pageName);
+      // Add the html2pdf script
+      await loadScript('https://cdnjs.cloudflare.com/ajax/libs/html2pdf.js/0.10.1/html2pdf.bundle.min.js', {
+        integrity: 'sha512-GsLlZN/3F2ErC5ifS5QtgpiJtWd43JWSuIgh7mbzZ8zBps+dvLusV+eNQATqgA/HdeKFVgA5v3S/cIrLF7QnIg==',
+        crossorigin: 'anonymous',
+        referrerpolicy: 'no-referrer',
+      });
+      if (window.html2pdf) {
+        await generatePDF(pageTitle);
```
Comment (ravkiran) [inline blocks/aside/aside.js:169]: @amol-anand `pageTitle` may have special chars that are not allowed in file system, can we use pageName here?

**Comment by ravkiran** [comment]
@amol-anand this output looks cleaner than the jsPdf. 
Few issues noticed
1. text is not selectable, shows like an image in pdf (text is selectable in jsPdf)
2. links are not clickable (same issue in jsPdf as well)

File: blocks/aside/aside.js (hunk 1 above)
Comment (amol-anand) [reply blocks/aside/aside.js:169]: good point. yes will update to pageName instead.

**Comment by amol-anand** [comment]
> @amol-anand this output looks cleaner than the jsPdf. Few issues noticed
> 
> 1. text is not selectable, shows like an image in pdf (text is selectable in jsPdf)
> 2. links are not clickable (same issue in jsPdf as well)

1. text is not selectable in html2pdf. known issue. Not sure if this is a bigger problem or clickable links is a bigger issue? For me it is the latter. if someone wants to print it, they can easily print this pdf or share it. 
2. links are clickable and work just fine for me (see screenshot)
Screenshot 2023-10-15 at 6 40 27 PM

**Comment by jjjaspher** [comment]
Hi @amol-anand, the only issue I've noticed is the text is not selectable. Main issue of links are not clickable and misaligned is resolved.

**Pull request merged**